### DISCUSSION TITLE
How to receive `event` parameters from the front-end?

### DISCUSSION THREAD

**Discussion by Vzz1c** [tauri-apps/tauri#7220] [Q&A]
```js
const test = ref("")
async function greet(event) {
  test.value= await invoke("greet", { name: event});
}
window.__TAURI__.event.listen('tauri://file-drop', function (event) {
  greet(event)
})
```
------------------------
```rust
#[tauri::command]
fn greet(name: ???????) -> Result<(),Error> {
    println!("12122");
    let id = app.listen_global("tauri://file-drop", |event| {
        println!("got event-name with payload {:?}", event.payload());
    });
    // unlisten to the event using the `id` returned on the `listen_global` function
    // a `once_global` API is also exposed on the `App` struct
    app.unlisten(id);

    // emit the `event-name` event to all webview windows on the frontend
    app.emit_all("tauri://file-drop", Payload { message: "Tauri is awesome!".into() }).unwrap();
    Ok(())
}
```
What type should I use to receive it? (I know I can just receive files[0] as a string, but it's not what I want)

**Comment by FabianLars** [comment]
`tauri://` events only get emitted to the frontend. On the rust side you'll have to use the explicit event handlers, in this case on_window_event whose event type has a [FileDrop](https://docs.rs/tauri/latest/tauri/enum.WindowEvent.html#variant.FileDrop) variant.

**Reply by FabianLars** [reply]
Or maybe i misunderstood the issue because you're trying to listen to the same event in your command.
A more general response here would be to use `Vec<String>` or `Vec<PathBuf>`

**Reply by Vzz1c** [reply]
`Vec<PathBuf>` can be used, I also see enumerations that may be relevant
```rust
pub enum FileDropEvent {
    Hovered(Vec<PathBuf, Global>).
    Dropped(Vec<PathBuf, Global>).
    Cancelled.
}
```
Is it possible to use this enumeration here? If not, under what circumstances does it work? I'm not very good at rust, it would be nice to have an example, I think I should be able to understand it, thanks

**Reply by FabianLars** [reply]
In this case i'd look at the externally or internally tagged representations: <https://serde.rs/enum-representations.html> and make sure your Payload is in the expected format.

Edit: _But_ i think you can't derive Deserialize for Tauri's FileDropEvent so you'd have to duplicate that enum in your code. Tauri's enum is primarily meant for use in the filedrop apis i mentioned in my first message.
-> I didn't get that you were talking about Tauri's enum at first, sorry.
